aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -145,26 +149,42 @@ static void upsample_nearest2d_out_cuda_template(
       input_height > 0 && input_width > 0 && output_height > 0 &&
       output_width > 0);
 
-  output.resize_({input.size(0), input.size(1), output_height, output_width});
+  Tensor input = input_.contiguous();
+  output.resize_({nbatch, channels, output_height, output_width});
   output.zero_();
 
-  const int num_kernels = output_height * output_width;
-  const int num_threads = std::min(
-      at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
-  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+  int nc = nbatch * channels;
+
+  const int max_threads = std::min<int>(
+      at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, MAX_THREADS);
+
+  int block_x = std::min<int>(lastPow2(output_width), max_threads);
+  int block_y = std::min<int>(lastPow2(output_height), max_threads / block_x);
+  int block_z = std::min<int>(nc, max_threads / block_x / block_y);
```

Comment:
Are you sure block_x, block_y, block_z can never be zero?

Reply:
In this case as output_width/height != 0, no they would not be zero.
I'll put a comment here.